### ISSUE TITLE
[Regression] 2.9.0 warns about proto deprecated even if proto:true

### ISSUE THREAD

**Issue by Krinkle** [jshint/jshint#2656]
```js
function foo( obj, key, val, injectFn ) {
	var spy = function () {
		injectFn();
		return val.apply( this, arguments );
	};

	// (..)
	/*jshint proto:true */
	spy.__proto__ = val;

	obj[ key ] = spy;
}
foo();
```

```
foo.js: line 9, col 26, The '__proto__' property is deprecated.
```

This worked fine on v2.8.0 and still works fine on http://jshint.com.

This is causing continuous integration problems because grunt-contrib-jshint silently upgraded from 2.8.0 to 2.9.0, thus causing all builds to fail.

**Comment by lukeapage** [comment]
is it just with inline options in this code? 
it could be https://github.com/jshint/jshint/issues/2471 which isn't a regression, but something may make it happen with the above code in 2.9 which doesn't  in 2.8

If so I have an open pr to fix it (no computer near to test the hypothesis)

**Comment by Krinkle** [comment]
I've narrowed it down a bit. It depends on context, seems parsing related? When the function is defined as method in an object literal it fails. When the function assigned as property directly it passes.

```js
// Fails in 2.9.0
// Passes in 2.8.0
( function () {
	function CompletenessTest() {
		return this;
	}

	CompletenessTest.prototype = {
		injectCheck: function ( spy, val ) {
			/*jshint proto:true */
			spy.__proto__ = val;
		}
	};

}() );
```
```js
// Passes in 2.9.0
// Passes in 2.8.0
( function () {
	function CompletenessTest() {
		return this;
	}

	CompletenessTest.injectCheck = function ( spy, val ) {
		/*jshint proto:true */
		spy.__proto__ = val;
	};

}() );
```

**Comment by jugglinmike** [comment]
Thanks for the report, @Krinkle! As an immediate fix, [we've unpublished 2.9.0](https://twitter.com/JSHint/status/639564453852356608), but a slightly-longer term (i.e. 2.9.1) fix is on the way: gh-2664

**Comment by Krinkle** [comment]
Thanks. Meanwhile, I figured out a workaround by turning jshint off for those lines. It seems that the `ignore` directive does work, where `proto` does not.

```diff
+			/*jshint ignore:start */
 			/*jshint proto:true */
 			spy.__proto__ = val;
+			/*jshint ignore:end */
```

**Comment by lukeapage** [comment]
I have merged the fix into master.

**Comment by Krinkle** [comment]
Fixed in ceca549861fb027e12fe5a2e05241b45ec337d7d.
